From 158a429154af39896a7c6edcd0d07e83b2a4368e Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Mon, 13 Nov 2017 15:00:04 +0100
Subject: [PATCH 2/3] build: fix win32 build

Remove msvcr100.dll dependency. This dll  is not avaible on old Windows
versions. Don't set locale on Windows, this doesn't seem to have any
consequences with VLC on windows (tested with US/FR locales).
---
 src/osdep/printf_useloc_win.c | 18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/src/osdep/printf_useloc_win.c b/src/osdep/printf_useloc_win.c
index 0c4c3be..fc56f90 100644
--- a/src/osdep/printf_useloc_win.c
+++ b/src/osdep/printf_useloc_win.c
@@ -16,43 +16,37 @@
  */
 
 #include <stdlib.h>
-#include <locale.h>
 
 #include "osdep/printf.h"
-
-static _locale_t cloc;
+#include <locale.h>
+#include <windows.h>
 
 void printf_c_init()
 {
-    cloc = _create_locale(LC_ALL, "C");
-    if (!cloc)
-        abort();
 }
 
 void printf_c_uninit()
 {
-    _free_locale(cloc);
-    cloc = (_locale_t) 0;
 }
 
 int vprintf_c(const char *format, va_list ap)
 {
-    return _vprintf_l(format, cloc, ap);
+    return vprintf(format, ap);
 }
 
 int vfprintf_c(FILE *stream, const char *format, va_list ap)
 {
-    return _vfprintf_l(stream, format, cloc, ap);
+    return vfprintf(stream, format, ap);
 }
 
 int vsprintf_c(char *str, const char *format, va_list ap)
 {
-    return _vsprintf_l(str, format, cloc, ap);
+    return vsprintf(str, format, ap);
 }
 
 int vsnprintf_c(char *str, size_t size, const char *format, va_list ap)
 {
-    return _vsnprintf_l(str, size, format, cloc, ap);
+    return vsnprintf(str, size, format, ap);
 }
 
 #define WRAP(fn, ...)                               \
-- 
2.11.0

